src/shared_modules/keystore/src/keyStore.cpp

```diff
@@ -0,0 +1,132 @@
+#include "keyStore.hpp"
+#include "evpHelper.hpp"
+#include "loggerHelper.h"
+#include "rocksDBWrapper.hpp"
+#include "rsaHelper.hpp"
+
+// Database constants, based on the keystore path.
+constexpr auto DATABASE_PATH {"queue/keystore"};
+
+// File constants, used in the version 1 of the keystore, currently in use only for the upgrade to version 2, that uses
+// AES 256 encryption, without the need of the private key file.
+constexpr auto PRIVATE_KEY_FILE {"etc/sslmanager.key"};
+
+// Keystore constants, KS_NAME is used in the logs as tag.
+// KS_VERSION is the current version of the keystore. Used to identify the version of the keystore in the database.
+// KS_VERSION_FIELD is the field used to store the version of the keystore in the database.
+constexpr auto KS_NAME {"keystore"};
+constexpr auto KS_VERSION {"2"};
+constexpr auto KS_VERSION_FIELD {"version"};
+
+static void upgrade(Utils::RocksDBWrapper& keystoreDB, const std::string& columnFamily)
+{
+    std::string versionValue;
+
+    // If the version field does not exist, it means that the keystore has not been upgraded yet.
+    if (!keystoreDB.get(KS_VERSION_FIELD, versionValue, columnFamily))
+    {
+        try
+        {
+            std::string rawValue;
+
+            // Check if the private key file exists
+            if (!std::filesystem::exists(PRIVATE_KEY_FILE))
+            {
+                throw std::runtime_error("Private key file not found.");
+            }
+
+            // Upgrade all keys
+            for (const auto& [key, value] : keystoreDB.begin(columnFamily))
+            {
+                std::string encryptedRSAValue;
+                std::vector<char> encryptedValue;
+
+                // Get the encrypted RSA value
+                if (keystoreDB.get(key, encryptedRSAValue, columnFamily))
+                {
+                    logInfo(KS_NAME, "Upgrading '%s' key pair.", key.c_str());
+                }
+
+                // Decrypt the RSA value
+                RSAHelper().rsaDecrypt(PRIVATE_KEY_FILE, encryptedRSAValue, rawValue);
+
+                // Encrypt the value with AES 256
+                EVPHelper().encryptAES256(rawValue, encryptedValue);
+
+                // Insert the key-value pair using AES encryption
+                keystoreDB.put(key, rocksdb::Slice(encryptedValue.data(), encryptedValue.size()), columnFamily);
+
+                logInfo(KS_NAME, "Key pair '%s' upgraded.", key.c_str());
+            }
+        }
+        catch (const std::exception& exception)
+        {
+            // If the upgrade fails, delete all keys and log the error.
+            keystoreDB.deleteAll(columnFamily);
+            logWarn(KS_NAME,
+                    "Keystore upgrade failed, re-run the tool again for all keys to save. Error: %s",
+                    exception.what());
```

Comment:
The message is not clear enough

```suggestion
            logWarn(KS_NAME,
                    "Keystore upgrade failed. Re-run the tool to regenerate the deleted keys in column '%s'. Error: %s",
                    columnFamily,
                    exception.what());
```

OR: 

```suggestion
            logWarn(KS_NAME,
                    "Keystore upgrade failed, re-run the tool again for all keys to save them. Error: %s",
                    columnFamily,
                    exception.what());```